File: poo-taller-1/Estudiante.java
import java.util.ArrayList;

public class Estudiante {
    // Atributos privados (encapsulación)
    private String nombre;
    private int edad;
    private String carrera;
    private ArrayList<String> materias;

    // Constructor
    public Estudiante(String nombre, int edad, String carrera) {
        this.nombre = nombre;
        this.edad = edad;
        this.carrera = carrera;
        this.materias = new ArrayList<>();
    }

    // Método para inscribir materia
    public void inscribirMateria(String materia) {
        this.materias.add(materia);
        System.out.println(nombre + " se inscribió en " + materia);
    }

    // Método para presentarse
    public void presentarse() {
        System.out.println("Hola, soy " + nombre);
        System.out.println("Tengo " + edad + " años");
        System.out.println("Estudio " + carrera);
        System.out.println("Materias inscritas: " + materias.size());
    }

    //Metodo para estudiar
    public void estudiar(int horas) { 
        System.out.println(nombre + " estudió " + horas + " horas"); 
    } 

    // Método main para probar
    public static void main(String[] args) {
        // Crear objetos
        Estudiante est3 = new Estudiante("Santiago Hernandez", 21, "Tec. Sist. Datos");

        // Usar métodos
        System.out.println("---");
        est3.inscribirMateria("POO");
        est3.inscribirMateria("Calculo 2");
        est3.inscribirMateria("Estructuras de datos");
        est3.inscribirMateria("Textos 2");
        est3.inscribirMateria("Fisica 1");
        est3.inscribirMateria("Catedra contexto");
        est3.presentarse();
        est3.estudiar(10);
    }
}

File: PYTHON/poo-taller-1/estudiante.py
class Estudiante:
    # Constructor - se ejecuta al crear un objeto
    def __init__(self, nombre, edad, carrera):
        self.nombre = nombre
        self.edad = edad
        self.carrera = carrera
        self.materias = []

    # Método para agregar materias
    def inscribir_materia(self, materia):
        self.materias.append(materia)
        print(f'{self.nombre} se inscribió en {materia}')

    # Método para mostrar información
    def presentarse(self):
        print(f'Hola, soy {self.nombre}')
        print(f'Tengo {self.edad} años')
        print(f'Estudio {self.carrera}')
        print(f'Materias inscritas: {len(self.materias)}')
        
        
    def estudiar(self, horas): 
        print(f'{self.nombre} estudió {horas} horas')

# --- Crear objetos (instancias de la clase) ---

estudiante1 = Estudiante('Santiago Hernandez', 21, 'Tec. Sist. de datos')

# --- Usar los métodos ---

estudiante1.presentarse()
print('---')
estudiante1.inscribir_materia('POO')
estudiante1.inscribir_materia('Calculo Integral')
estudiante1.inscribir_materia('Fisica Newtoniana')
print('---')
estudiante1.estudiar(5)


